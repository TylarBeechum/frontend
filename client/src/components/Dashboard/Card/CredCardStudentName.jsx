import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const CredCardStudentName = ({ ownerName }) => {
  return (
    <StudentNameContainer>
      <h3>{ownerName}</h3>
    </StudentNameContainer>
  );
};

CredCardStudentName.propTypes = {
  ownerName: PropTypes.string.isRequired
};

const StudentNameContainer = styled.div`
  display: flex;
  align-items: center;
  width: 20%;
`;

export default CredCardStudentName;
