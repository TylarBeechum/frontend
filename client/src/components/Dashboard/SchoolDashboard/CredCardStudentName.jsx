import React from 'react'

const CredCardStudentName = () => {
  return (
    <dir>
        <h3>Cecil John Tantay</h3>
      </dir>
  )
}

export default CredCardStudentName
