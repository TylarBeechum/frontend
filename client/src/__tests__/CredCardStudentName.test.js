/* eslint-disable */
import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import {
  renderWithRouterAndProviders,
  rendererWithRouterAndProviders
} from '../testHelpers';

import CredCardStudentName from '../components/Dashboard/Card/CredCardStudentName';

describe('<CredCardStudentName />', () => {
  it('should render to the screen without crashing', () => {
    const helpers = renderWithRouterAndProviders(
      <CredCardStudentName ownerName="testing" />
    );
  });

  it('should match the previous snapshot of the component', () => {
    const tree = rendererWithRouterAndProviders(
      <CredCardStudentName ownerName="testing" />
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should render the owner name successfully', () => {
    const { getByText } = renderWithRouterAndProviders(
      <CredCardStudentName ownerName="testing" />
    );
    const ownerName = getByText(/testing/i);
    expect(ownerName).toBeDefined();
  });
});
