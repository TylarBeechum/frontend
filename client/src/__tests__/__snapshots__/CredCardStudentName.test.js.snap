// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CredCardStudentName /> should match the previous snapshot of the component tests that the component has not changed and matched the previous snapshot 1`] = `
<div
  className="StyledGrommet-sc-19lkkz7-0 iDwiuh"
>
  <div
    className="sc-EHOje eSogAa"
  >
    <h3>
      testing
    </h3>
  </div>
</div>
`;
